drivers/spi/spi_litespi.c

```diff
@@ -40,7 +40,8 @@ static int spi_config(const struct spi_config *config, uint16_t *control)
 		return -ENOTSUP;
 	}
 
-	if ((config->operation & SPI_LINES_MASK) != SPI_LINES_SINGLE) {
+	if (IS_ENABLED(CONFIG_SPI_EXTENDED_MODES) &&
+	    (config->operation & SPI_LINES_MASK) != SPI_LINES_SINGLE) {
```

Comment:
nit: would be nice to provide some helpers, e.g.

```c
#define SPI_GET_LINES(operation) ((operation) & SPI_LINES_MASK)
```

to then do:

```c
SPI_GET_LINES(config->operation) != SPI_LINES_SINGLE) {
```

Reply:
Let's no that in #39991